File: dSoak-Common-Java/src/SharedObjectTester/ProcessDataTester.java
package SharedObjectTester;

import static org.junit.Assert.*;

import org.junit.Test;

import SharedObject.ProcessData;

public class ProcessDataTester
{
	@Test
	public void test_Everything() 
	{
		 ProcessData data = new ProcessData();
		 
		 assertEquals(0, data.GameId);
		 assertEquals(0, data.ProcessId);
		 assertEquals(null, data.ProcessType);
		 assertEquals(0, data.LifePoints);
		 assertEquals(0, data.HitPoints);
		 assertEquals(0, data.NumberOfPennies);
		 assertEquals(0, data.NumberOfUnfilledBalloon);
		 assertEquals(0, data.NumberOfFilledBalloon);
		 assertEquals(0, data.NumberOfUnraisedUmbrellas);
		 assertFalse(data.HasUmbrellaRaised);

         data = new ProcessData();
         data.GameId = 10;
         data.ProcessId = 11;
         data.ProcessType = ProcessData.PossibleProcessType.Thief;
         data.LifePoints = 12;
         data.HitPoints = 13;
         data.NumberOfPennies = 14;
         data.NumberOfUnfilledBalloon = 15;
         data.NumberOfFilledBalloon = 16;
         data.NumberOfUnraisedUmbrellas = 17;
         data.HasUmbrellaRaised = true;
         

         assertEquals(10, data.GameId);
         assertEquals(11, data.ProcessId);
         assertEquals(ProcessData.PossibleProcessType.Thief, data.ProcessType);
         assertEquals(12, data.LifePoints);
         assertEquals(13, data.HitPoints);
         assertEquals(14, data.NumberOfPennies);
         assertEquals(15, data.NumberOfUnfilledBalloon);
         assertEquals(16, data.NumberOfFilledBalloon);
         assertEquals(17, data.NumberOfUnraisedUmbrellas);
         assertTrue(data.HasUmbrellaRaised);
	}

}

File: dSoak-Common-CSharp/SharedObjectTesting/ProcessDataTester.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

using SharedObjects;

namespace SharedObjectTesting
{
    [TestClass]
    public class ProcessDataTester
    {
        [TestMethod]
        public void ProcessData_TestEverything()
        {
            ProcessData data = new ProcessData();
            Assert.AreEqual(0, data.GameId);
            Assert.AreEqual(0, data.ProcessId);
            Assert.AreEqual(ProcessData.PossibleProcessType.Unknown, data.ProcessType);
            Assert.AreEqual(0, data.LifePoints);
            Assert.AreEqual(0, data.HitPoints);
            Assert.AreEqual(0, data.NumberOfPennies);
            Assert.AreEqual(0, data.NumberOfUnfilledBalloon);
            Assert.AreEqual(0, data.NumberOfFilledBalloon);
            Assert.AreEqual(0, data.NumberOfUnraisedUmbrellas);
            Assert.IsFalse(data.HasUmbrellaRaised);

            data = new ProcessData()
            {
                GameId = 10,
                ProcessId = 11,
                ProcessType = ProcessData.PossibleProcessType.Thief,
                LifePoints = 12,
                HitPoints = 13,
                NumberOfPennies = 14,
                NumberOfUnfilledBalloon = 15,
                NumberOfFilledBalloon = 16,
                NumberOfUnraisedUmbrellas = 17,
                HasUmbrellaRaised = true
            };

            Assert.AreEqual(10, data.GameId);
            Assert.AreEqual(11, data.ProcessId);
            Assert.AreEqual(ProcessData.PossibleProcessType.Thief, data.ProcessType);
            Assert.AreEqual(12, data.LifePoints);
            Assert.AreEqual(13, data.HitPoints);
            Assert.AreEqual(14, data.NumberOfPennies);
            Assert.AreEqual(15, data.NumberOfUnfilledBalloon);
            Assert.AreEqual(16, data.NumberOfFilledBalloon);
            Assert.AreEqual(17, data.NumberOfUnraisedUmbrellas);
            Assert.IsTrue(data.HasUmbrellaRaised);
        }
    }
}
